### DISCUSSION TITLE
How can I only keep data in one xarray for entries that match a specific dimension in another xarray?

### DISCUSSION THREAD

**Discussion by benoitespinola** [pydata/xarray#7567] [Q&A] [answered]
I have two datasets:
ds1, with coordinates time, lat, lon, depth.
ds2, with coordinates time.

The time in ds1 has a higher resolution than ds2 and all the times in ds2 are present in ds1.

For instance:
```
ds1.time = [0, 1, 2 , 3, 4, 5, 6, 7, 8, 9 ,10]
ds2.time = [0, 5, 10]
```
(I am using int instead of datetime64 here for the readability)

I would like to drop all the entries in ds1 where the time is NOT in ds2.time so that I only keep the records in ds1 such as:
`ds1.time = [0, 5, 10]`

How can I do that?

I tried sel, isel, resample, where... All failed.

**Comment by keewis** [comment]
if I understand correctly, you should be able to do `ds1.sel(time=ds2.time)`.

If that doesn't work, maybe your times are a little bit off? Try `method="nearest"` with a appropriate `tolerance` in that case.

Otherwise we'd need more information on your time variable and the code you tried.

**Reply by benoitespinola** [reply, marked as answer]
After a lot of research, trial and error, I got it to work as follow:

`ds1 = ds1.where(ds1.time.isin(ds2.time), drop=True)`
